fastapi/dependencies/utils.py

```diff
@@ -427,6 +435,49 @@ def is_coroutine_callable(call: Callable) -> bool:
     return values, errors, background_tasks, response, dependency_cache
 
 
+async def solve_generator(
+    call: Callable, sub_values: Dict[str, Any], background_tasks: BackgroundTasks
+) -> Any:
+    generator = call(**sub_values)
+
+    def solve() -> Any:
+        try:
+            return next(generator)
+        except StopIteration:
+            raise RuntimeError("generator didn't yield")
+
+    def teardown() -> None:
+        try:
+            next(generator)
+        except StopIteration:  # pragma: no cover
+            pass  # prevents BackgroundTasks from hanging
+        else:
+            raise RuntimeError("generator didn't stop")
```

Comment:
Note: this is precisely the same error message you'd get if `call` had been used in a `with` statement after decorating with `contextlib.contextmanager`.

Reply:
The same is the case for the `RuntimeError` above and those in the `solve_asyncgen` call below.